### PR TITLE
[FIX] Report Stock Quantity with multi location

### PR DESCRIPTION
**Pull request by sla-subteno-it**
Description of the issue/feature this PR addresses:
Creation of a warehouse with sub-level of locations and put the location of the warehouse in view type.

Current behavior before PR:
Make a purchase or sale order and the report will not return any values.

Desired behavior after PR is merged:
After this patch, the report return values.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by sle-odoo** [comment]
@amoyaux fyi

**Comment by amoyaux** [comment]
It's working at my end. I need more information in order to reproduce it.

**Comment by sla-subteno-it** [comment]
@amoyaux Why close it ? I open ticket in Enterprise contract.

**Comment by amoyaux** [comment]
I missed the part about PO/SO

**Comment by amoyaux** [comment]
The patch seems good. Can you add your CLA?

**Comment by sla-subteno-it** [comment]
@amoyaux CLA updated

**Comment by amoyaux** [comment]
I used another PR in order to rewrite commit message https://github.com/odoo/odoo/pull/44548

**Comment by sla-subteno-it** [comment]
@amoyaux Many thanks.

File: addons/stock/report/report_stock_quantity.py [hunk 1]
```diff
@@ -89,15 +88,15 @@ def init(self):
         ELSE date::date - interval '1 day'
     END, '1 day'::interval)::date date,
     CASE
-        WHEN (ls.usage = 'internal' OR ls.usage = 'transit' AND ls.company_id IS NOT NULL) AND ld.usage != 'internal' AND m.state = 'done' THEN product_qty
-        WHEN ls.usage != 'internal' AND (ld.usage = 'internal' OR ld.usage = 'transit' AND ld.company_id IS NOT NULL) AND m.state = 'done' THEN -product_qty
-        WHEN (ls.usage = 'internal' OR ls.usage = 'transit' AND ls.company_id IS NOT NULL) AND ld.usage != 'internal' THEN -product_qty
-        WHEN ls.usage != 'internal' AND (ld.usage = 'internal' OR ld.usage = 'transit' AND ld.company_id IS NOT NULL) THEN product_qty
+        WHEN (whs.id IS NULL AND whd.id IS NOT NULL) OR ld.usage = 'transit' AND m.state = 'done' THEN product_qty
```
Comment (amoyaux) [inline addons/stock/report/report_stock_quantity.py]: @sla-subteno-it I don't know if you applied it on some servers but there is a small mistake here. It shoud be whs.id IS NOT NULL AND whd.id IS NULL. Also the opposite bellow

**Comment by amoyaux** [comment]
Also I will remove the check on transit. I don't think they remain usefull

File: addons/stock/report/report_stock_quantity.py (hunk 1 above)
Comment (sla-subteno-it) [reply addons/stock/report/report_stock_quantity.py]: > Also I will remove the check on transit. I don't think they remain useful



> Also I will remove the check on transit. I don't think they remain useful

If we will transfer a stock between warehouses, the outgoing and incoming will not see in report stock.

Comment (sla-subteno-it) [reply addons/stock/report/report_stock_quantity.py]: > @sla-subteno-it I don't know if you applied it on some servers but there is a small mistake here. It shoud be whs.id IS NOT NULL AND whd.id IS NULL. Also the opposite bellow

Yes, I applied in server. Possible a mistake, I'll check it.

Comment (amoyaux) [reply addons/stock/report/report_stock_quantity.py]: > > Also I will remove the check on transit. I don't think they remain useful
> 
> If we will transfer a stock between warehouses, the outgoing and incoming will not see in report stock.

Okay agree
@oma-odoo

**Comment by amoyaux** [comment]
merged in https://github.com/odoo/odoo/pull/44548

**Pull request closed without merging**